From 15bf84f69221f6c8d1d147909f675765b531536f Mon Sep 17 00:00:00 2001
From: Francois Daoust <fd@tidoust.net>
Date: Thu, 22 Feb 2024 14:40:00 +0100
Subject: [PATCH] Add obsolete HTML elements

Patches the list of HTML elements with elements that are entirely obsolete but
still need to be supported by browsers, see:
https://html.spec.whatwg.org/multipage/obsolete.html#non-conforming-features

These obsolete elements cannot easily be extracted automatically. The list is
unlikely to evolve significantly over time, and should only grow as new elements
get obsoleted. When that happens, the element would first disappear from
`elements/html.json`, which should provide the nudge we need to update this
patch.

Obsolete element definitions come with an `"obsolete": true` property to
distinguish them from regular HTML elements. Some of them have dedicated
interfaces, and a couple of others inherit from specific interfaces (this is not
very clear in the spec though, e.g. `xmp` implements the `HTMLPreElement`
interface whereas `plaintext` does not, yet the prose in the spec is similar in
both cases).
---
 ed/elements/html.json | 174 ++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 174 insertions(+)

diff --git a/ed/elements/html.json b/ed/elements/html.json
index 177e528f2..e5924f279 100644
--- a/ed/elements/html.json
+++ b/ed/elements/html.json
@@ -563,6 +563,180 @@
       "name": "canvas",
       "href": "https://html.spec.whatwg.org/multipage/canvas.html#canvas",
       "interface": "HTMLCanvasElement"
+    },
+    {
+      "name": "applet",
+      "interface": "HTMLUnknownElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#applet",
+      "obsolete": true
+    },
+    {
+      "name": "acronym",
+      "interface": "HTMLElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#acronym",
+      "obsolete": true
+    },
+    {
+      "name": "bgsound",
+      "interface": "HTMLUnknownElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#bgsound",
+      "obsolete": true
+    },
+    {
+      "name": "dir",
+      "interface": "HTMLDirectoryElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#dir",
+      "obsolete": true
+    },
+    {
+      "name": "frame",
+      "interface": "HTMLFrameElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#frame",
+      "obsolete": true
+    },
+    {
+      "name": "frameset",
+      "interface": "HTMLFrameSetElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#frameset",
+      "obsolete": true
+    },
+    {
+      "name": "noframes",
+      "interface": "HTMLElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#noframes",
+      "obsolete": true
+    },
+    {
+      "name": "isindex",
+      "interface": "HTMLUnknownElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#isindex",
+      "obsolete": true
+    },
+    {
+      "name": "keygen",
+      "interface": "HTMLUnknownElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#keygen",
+      "obsolete": true
+    },
+    {
+      "name": "listing",
+      "interface": "HTMLPreElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#listing",
+      "obsolete": true
+    },
+    {
+      "name": "menuitem",
+      "interface": "HTMLElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#menuitem",
+      "obsolete": true
+    },
+    {
+      "name": "nextid",
+      "interface": "HTMLUnknownElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#nextid",
+      "obsolete": true
+    },
+    {
+      "name": "noembed",
+      "interface": "HTMLElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#noembed",
+      "obsolete": true
+    },
+    {
+      "name": "param",
+      "interface": "HTMLParamElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#param",
+      "obsolete": true
+    },
+    {
+      "name": "plaintext",
+      "interface": "HTMLElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#plaintext",
+      "obsolete": true
+    },
+    {
+      "name": "rb",
+      "interface": "HTMLElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#rb",
+      "obsolete": true
+    },
+    {
+      "name": "rtc",
+      "interface": "HTMLElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#rtc",
+      "obsolete": true
+    },
+    {
+      "name": "strike",
+      "interface": "HTMLElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#strike",
+      "obsolete": true
+    },
+    {
+      "name": "xmp",
+      "interface": "HTMLPreElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#xmp",
+      "obsolete": true
+    },
+    {
+      "name": "basefont",
+      "interface": "HTMLElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#basefont",
+      "obsolete": true
+    },
+    {
+      "name": "big",
+      "interface": "HTMLElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#big",
+      "obsolete": true
+    },
+    {
+      "name": "blink",
+      "interface": "HTMLUnknownElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#blink",
+      "obsolete": true
+    },
+    {
+      "name": "center",
+      "interface": "HTMLElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#center",
+      "obsolete": true
+    },
+    {
+      "name": "font",
+      "interface": "HTMLFontElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#font",
+      "obsolete": true
+    },
+    {
+      "name": "marquee",
+      "interface": "HTMLMarqueeElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#the-marquee-element",
+      "obsolete": true
+    },
+    {
+      "name": "multicol",
+      "interface": "HTMLUnknownElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#multicol",
+      "obsolete": true
+    },
+    {
+      "name": "nobr",
+      "interface": "HTMLElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#nobr",
+      "obsolete": true
+    },
+    {
+      "name": "spacer",
+      "interface": "HTMLUnknownElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#spacer",
+      "obsolete": true
+    },
+    {
+      "name": "tt",
+      "interface": "HTMLElement",
+      "href": "https://html.spec.whatwg.org/multipage/obsolete.html#tt",
+      "obsolete": true
     }
   ]
 }
\ No newline at end of file
-- 
2.42.0.windows.2

